### ISSUE TITLE
AWS volume attached but not mounted, service started anyway

### ISSUE THREAD

**Issue by george-angel** [kubernetes/kubernetes#32295]
OS: CoreOS stable (1068.10.0)
Kubelet: quay.io/coreos/hyperkube:v1.3.6_coreos.0

Interesting issue, I have a prometheus deployment with a PVC ebs volume - 400GB.

The volume is attached but is not mounted:

```
core@ip-10-66-31-165 ~ $ lsblk
NAME    MAJ:MIN   RM  SIZE RO TYPE MOUNTPOINT
xvda    202:0      0   52G  0 disk
|-xvda1 202:1      0  128M  0 part /boot
|-xvda2 202:2      0    2M  0 part
|-xvda3 202:3      0    1G  0 part /usr
|-xvda4 202:4      0    1G  0 part
|-xvda6 202:6      0  128M  0 part /usr/share/oem
|-xvda7 202:7      0   64M  0 part
`-xvda9 202:9      0 49.7G  0 part /
xvdba   202:13312  0   20G  0 disk /var/lib/kubelet/pods/3c32c00a-75b0-11e6-aa52-063165fd7393/volumes/kubernetes.io~aws-ebs/pvc-f884ac4b-5d51-11e6-94d4-02fe9eee2d51
xvdbb   202:13568  0  400G  0 disk
```

`df` on the machine

```
core@ip-10-66-31-165 ~ $ sudo df -khT
Filesystem     Type      Size  Used Avail Use% Mounted on
devtmpfs       devtmpfs  3.9G     0  3.9G   0% /dev
tmpfs          tmpfs     3.9G     0  3.9G   0% /dev/shm
tmpfs          tmpfs     3.9G  2.0M  3.9G   1% /run
tmpfs          tmpfs     3.9G     0  3.9G   0% /sys/fs/cgroup
/dev/xvda9     ext4       49G  9.0G   38G  20% /
/dev/xvda3     ext4      985M  624M  310M  67% /usr
/dev/xvda1     vfat      128M   38M   90M  30% /boot
tmpfs          tmpfs     3.9G     0  3.9G   0% /tmp
tmpfs          tmpfs     3.9G     0  3.9G   0% /media
/dev/xvda6     ext4      108M   64K   99M   1% /usr/share/oem
overlay        overlay    49G  9.0G   38G  20% /var/lib/rkt/pods/run/97963c99-2e18-4982-be42-d65e7c7e8265/stage1/rootfs
overlay        overlay    49G  9.0G   38G  20% /var/lib/rkt/pods/run/97963c99-2e18-4982-be42-d65e7c7e8265/stage1/rootfs/opt/stage2/hyperkube/rootfs
tmpfs          tmpfs     3.9G     0  3.9G   0% /var/lib/rkt/pods/run/97963c99-2e18-4982-be42-d65e7c7e8265/stage1/rootfs/opt/stage2/hyperkube/rootfs/tmp
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/902a52e1-75a9-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/902a1593-75a9-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/9034b7f9-75a9-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
/dev/xvdba     ext4       20G  7.8G   11G  42% /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/eu-west-1b/vol-be33854b
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/7f33be6f-75ad-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/19c0d8aa-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/19b686f6-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/19d4f7c8-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/1a07819f-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/19efc7dc-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/1a24c222-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/55c20d83-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/5610f25c-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-3l2jv
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/56350769-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-0uiot
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/562610aa-75ae-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-7eb11
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/3c32c00a-75b0-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G   12K  3.9G   1% /var/lib/kubelet/pods/3c606fde-75b0-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/default-token-zon4r
tmpfs          tmpfs     3.9G  4.0K  3.9G   1% /var/lib/kubelet/pods/3c606fde-75b0-11e6-aa52-063165fd7393/volumes/kubernetes.io~secret/uw-service-billing-secrets-volume
tmpfs          tmpfs     799M     0  799M   0% /run/user/500
```

And I can see some failed mount services:

```
Failed Units: 3
  var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-eu\x2dwest\x2d1b-vol\x2d7e79f78b.mount
  var-lib-kubelet-pods-0c294ad5\x2d75aa\x2d11e6\x2daa52\x2d063165fd7393-volumes-kubernetes.io\x7eaws\x2debs-pvc\x2df884ac4b\x2d5d51\x2d11e6\x2d94d4\x2d02fe9eee2d51.mount
  var-lib-rkt-pods-run-97963c99\x2d2e18\x2d4982\x2dbe42\x2dd65e7c7e8265-stage1-rootfs-opt-stage2-hyperkube-rootfs-var-lib-kubelet-plugins-kubernetes.io-aws\x2debs-mounts-aws-eu\x2dwest\x2d1b-vol\x2d7e79f78b.mount
```

The interesting bit is that my pod is running anyway, and running `df` inside container:

```
system(master)/kubectl exec -ti prometheus-1394874088-32qhv -- df -khT
Filesystem           Type            Size      Used Available Use% Mounted on
overlay              overlay        48.1G      8.9G     37.1G  19% /
tmpfs                tmpfs           3.9G         0      3.9G   0% /dev
tmpfs                tmpfs           3.9G         0      3.9G   0% /sys/fs/cgroup
/dev/xvdba           ext4           19.6G      7.8G     10.8G  42% /prometheus
/dev/xvda9           ext4           48.1G      8.9G     37.1G  19% /etc/prometheus
/dev/xvda9           ext4           48.1G      8.9G     37.1G  19% /dev/termination-log
tmpfs                tmpfs           3.9G     12.0K      3.9G   0% /var/run/secrets/kubernetes.io/serviceaccount
/dev/xvda9           ext4           48.1G      8.9G     37.1G  19% /etc/resolv.conf
/dev/xvda9           ext4           48.1G      8.9G     37.1G  19% /etc/hostname
/dev/xvda9           ext4           48.1G      8.9G     37.1G  19% /etc/hosts
shm                  tmpfs          64.0M         0     64.0M   0% /dev/shm
```

This is the mount point:

`/dev/xvdba           ext4           19.6G      7.8G     10.8G  42% /prometheus`

I'm not actually sure what that mount is, another pod using pvc on the same host is mongo, and thats mounted there.

**Comment by jingxu97** [comment]
Doesn't the output below show that your aws ebs volume is mounted?
xvdba   202:13312  0   20G  0 disk /var/lib/kubelet/pods/3c32c00a-75b0-11e6-aa52-063165fd7393/volumes/kubernetes.io~aws-ebs/pvc-f884ac4b-5d51-11e6-94d4-02fe9eee2d51

/dev/xvdba ext4 19.6G 7.8G 10.8G 42% /prometheus is the global mount path, then for each pod, we have bind mount

**Comment by Zapadlo** [comment]
Thats a *different* EBS volume, which is 20G in size and belongs to Mongo. It happenned to be running on the same host.

**Comment by george-angel** [comment]
Something similar happened today, bu this time I got a lot more debugging done.

`df` inside the container:

```
k8s(calico-dev)/kubectl --context=dev -n system exec -ti prometheus-2255685456-ted2c -- df -khT
Filesystem           Type            Size      Used Available Use% Mounted on
overlay              overlay        48.1G      6.8G     39.2G  15% /
tmpfs                tmpfs           7.8G         0      7.8G   0% /dev
tmpfs                tmpfs           7.8G         0      7.8G   0% /sys/fs/cgroup
/dev/xvdba           ext4           19.6G     58.2M     18.5G   0% /prometheus
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /dev/termination-log
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /etc/prometheus
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /etc/kubernetes/ssl
tmpfs                tmpfs           7.8G     12.0K      7.8G   0% /var/run/secrets/kubernetes.io/serviceaccount
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /etc/resolv.conf
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /etc/hostname
/dev/xvda9           ext4           48.1G      6.8G     39.2G  15% /etc/hosts
shm                  tmpfs          64.0M         0     64.0M   0% /dev/shm
tmpfs                tmpfs           7.8G         0      7.8G   0% /proc/kcore
tmpfs                tmpfs           7.8G         0      7.8G   0% /proc/latency_stats
tmpfs                tmpfs           7.8G         0      7.8G   0% /proc/timer_stats
tmpfs                tmpfs           7.8G         0      7.8G   0% /proc/sched_debug
```

`lsblk` on the host:

```
core@ip-10-66-21-94 ~ $ lsblk
NAME    MAJ:MIN   RM  SIZE RO TYPE MOUNTPOINT
xvda    202:0      0   52G  0 disk
|-xvda1 202:1      0  128M  0 part /boot
|-xvda2 202:2      0    2M  0 part
|-xvda3 202:3      0    1G  0 part /usr
|-xvda4 202:4      0    1G  0 part
|-xvda6 202:6      0  128M  0 part /usr/share/oem
|-xvda7 202:7      0   64M  0 part
`-xvda9 202:9      0 49.7G  0 part /
xvdba   202:13312  0   20G  0 disk /var/lib/kubelet/pods/676badd7-bc9d-11e6-b0e0-06de0177d7df/volumes/kubernetes.io~aws-ebs/pvc-cbcb8975-96b6-11e6-9736-0a9625c0fe05
xvdbb   202:13568  0  400G  0 disk /var/lib/kubelet/pods/67202d2f-bc9d-11e6-b0e0-06de0177d7df/volumes/kubernetes.io~aws-ebs/pvc-a5ab5df1-a062-11e6-b594-02ac14e32ac3
```

kubelet logs:

```
core@ip-10-66-21-94 ~ $ journalctl -u kubelet | grep xvdbb
Dec 07 16:52:25 ip-10-66-21-94.eu-west-1.compute.internal kubelet-wrapper[1569]: I1207 16:52:25.866835    1569 operation_executor.go:1074] Controller successfully attached volume "kubernetes.io/aws-ebs/aws://eu-west-1b/vol-0c1ea88a" (spec.Name: "pvc-a5ab5df1-a062-11e6-b594-02ac14e32ac3") pod "67202d2f-bc9d-11e6-b0e0-06de0177d7df" (UID: "67202d2f-bc9d-11e6-b0e0-06de0177d7df") devicePath: "/dev/xvdbb"
Dec 07 16:52:25 ip-10-66-21-94.eu-west-1.compute.internal kubelet-wrapper[1569]: I1207 16:52:25.884993    1569 operation_executor.go:707] Entering MountVolume.WaitForAttach for volume "kubernetes.io/aws-ebs/aws://eu-west-1b/vol-0c1ea88a" (spec.Name: "pvc-a5ab5df1-a062-11e6-b594-02ac14e32ac3") pod "67202d2f-bc9d-11e6-b0e0-06de0177d7df" (UID: "67202d2f-bc9d-11e6-b0e0-06de0177d7df") DevicePath: "/dev/xvdbb"


core@ip-10-66-21-94 ~ $ journalctl -u kubelet | grep xvdba
Dec 07 16:52:21 ip-10-66-21-94.eu-west-1.compute.internal kubelet-wrapper[1569]: I1207 16:52:21.065585    1569 operation_executor.go:1074] Controller successfully attached volume "kubernetes.io/aws-ebs/aws://eu-west-1b/vol-5209aed4" (spec.Name: "pvc-cbcb8975-96b6-11e6-9736-0a9625c0fe05") pod "676badd7-bc9d-11e6-b0e0-06de0177d7df" (UID: "676badd7-bc9d-11e6-b0e0-06de0177d7df") devicePath: "/dev/xvdba"
Dec 07 16:52:21 ip-10-66-21-94.eu-west-1.compute.internal kubelet-wrapper[1569]: I1207 16:52:21.161231    1569 operation_executor.go:707] Entering MountVolume.WaitForAttach for volume "kubernetes.io/aws-ebs/aws://eu-west-1b/vol-5209aed4" (spec.Name: "pvc-cbcb8975-96b6-11e6-9736-0a9625c0fe05") pod "676badd7-bc9d-11e6-b0e0-06de0177d7df" (UID: "676badd7-bc9d-11e6-b0e0-06de0177d7df") DevicePath: "/dev/xvdba"
```

The 20G pod and the 400G pod respectively:

```
k8s(calico-dev)/kubectl --context=dev -n partner-portal get pod graph-2699674866-kicng -o yaml
...
  uid: 67202d2f-bc9d-11e6-b0e0-06de0177d7df


k8s(calico-dev)/kubectl --context=dev -n system get pod prometheus-2255685456-ted2c -o yaml
...
  uid: 676badd7-bc9d-11e6-b0e0-06de0177d7df
```

The pvcs:

```
k8s(calico-dev)/kubectl --context=dev -n system get pvc prometheus-ebs-pvc -o yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: '{"kind":"PersistentVolumeClaim","apiVersion":"v1","metadata":{"name":"prometheus-ebs-pvc","namespace":"system","creationTimestamp":null,"annotations":{"volume.beta.kubernetes.io/storage-class":"ebs-gp2"}},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"400Gi"}}},"status":{}}'
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-class: ebs-gp2
  creationTimestamp: 2016-10-20T11:17:32Z
  name: prometheus-ebs-pvc
  namespace: system
  resourceVersion: "6665076"
  selfLink: /api/v1/namespaces/system/persistentvolumeclaims/prometheus-ebs-pvc
  uid: cbcb8975-96b6-11e6-9736-0a9625c0fe05


k8s(calico-dev)/kubectl --context=dev -n partner-portal get pvc neo4j-ebs-pvc -o yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: '{"kind":"PersistentVolumeClaim","apiVersion":"v1","metadata":{"name":"neo4j-ebs-pvc","namespace":"partner-portal","creationTimestamp":null,"annotations":{"volume.beta.kubernetes.io/storage-class":"ebs-gp2"}},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"20Gi"}}},"status":{}}'
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-class: ebs-gp2
  creationTimestamp: 2016-11-01T18:40:22Z
  name: neo4j-ebs-pvc
  namespace: partner-portal
  resourceVersion: "10190613"
  selfLink: /api/v1/namespaces/partner-portal/persistentvolumeclaims/neo4j-ebs-pvc
  uid: a5ab5df1-a062-11e6-b594-02ac14e32ac3
```

And screenshots of volumes in AWS :

![2016-12-07-171405_1690x295_scrot](https://cloud.githubusercontent.com/assets/1801067/20979828/1c761a58-bca6-11e6-9ac9-cb19165f4bac.png)
![2016-12-07-171324_1693x312_scrot](https://cloud.githubusercontent.com/assets/1801067/20979866/336825ee-bca6-11e6-9ae7-8a92cc4e8cf1.png)

So, from all that, volume `vol-0c1ea88a` (20G) attached to `xvdba` - AWS sees this, `lsblk` on the hosts sees this, but kubelet decides `DevicePath: "/dev/xvdbb"` forwhatever reason...

**Comment by jingxu97** [comment]
@george-angel, which version of kubernetes you are using? We recently have a fix for this type of error #33796. Could you please check whether you have this fix in your setup? Thanks!

**Comment by george-angel** [comment]
@jingxu97 `v1.4.0_coreos.2` - hyperkube.

Thanks for that, I'll make sure to upgrade to the latest.
